### PR TITLE
Add Auth0 to provider list

### PR DESCRIPTION
**Pull request by btiernay**
## Summary
After addressing the [open issues](https://github.com/orgs/mcp-auth/discussions/33), this PR adds Auth0 to the documented list of identity providers tested [with MCP Auth](https://mcp-auth.dev/docs/provider-list).

What’s included:
- Updates to the Provider list page to include Auth0, with ✅ for dynamic client registration and ⚠️ for resource indicator support.
- Footnote explains that while Auth0 supports [multi-resource refresh tokens (MRRT)](https://learn.microsoft.com/en-us/entra/identity-platform/multi-resource-refresh-tokens), it does not support full RFC 8707 resource indicators.
- Verified that Auth0 passes all compatibility checks via the embedded `<TestProvider />` tool.
- Minor clarification added to the dynamic client registration FAQ to guide users based on their use case (internal vs. public MCP servers).

This helps users better evaluate which providers are plug-and-play with MCP Auth and what workarounds or limitations to expect.

Open to feedback or additional notes on edge-case behavior.

## Test plan
Verified provider is displayed correctly in the documentation:
image

And with the test tool:
image

### CODE REVIEW

File: docs/provider-list.mdx [hunk 1]
```diff
@@ -10,16 +10,19 @@ import TestProvider from '@site/src/components/TestProvider';
 This list contains providers that have been tested with MCP Auth.
 
 | Provider                             | Type           | OAuth 2.1 | Metadata URL | Dynamic Client Registration | Resource Indicator |
```
Comment (btiernay) [inline docs/provider-list.mdx:12]: @xiaoyijun Wasn't quite sure what Resource Indicator meant here myself. Perhaps adding some docs would be useful as well?

Comment (xiaoyijun) [reply docs/provider-list.mdx]: I have added the Resource Indicator rfc link to the table.

**Review by xiaoyijun — APPROVED** [review]

**Comment by xiaoyijun** [comment]
Hi @btiernay , merging is blocked because your commits do not have verified signatures.
Please see the instructions here to sign your commits:
https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification

**Comment by xiaoyijun** [comment]
Hi @btiernay , I’ve resolved the issue by re-signing the commits during a rebase, so merging is now unblocked.
Just as a heads-up, our main branch requires all commits to be signed. For future contributions, it would be great if you could set up commit signing as mentioned in the link in the previous comment.

File: docs/provider-list.mdx (hunk 1 above)
Comment (btiernay) [reply docs/provider-list.mdx]: @xiaoyijun Thanks! Can you explain why this is important to call out? I think linking the relevant parts of the MCP spec would be useful in this document as well.
